car.capnp

```diff
@@ -181,6 +184,11 @@ struct CarState {
       altButton1 @6;
       altButton2 @7;
       altButton3 @8;
+      setCruise @9;
+      resumeCruise @10;
+      gapAdjustCruise @11;
+      increaseCruise @12;
+      decreaseCruise @13;
```

Comment:
accelCruise and decelCruise are enough, aren't they?

Reply:
I'm afraid they aren't. The detailed reasoning is in the wall of text above, but TL;DR:

> This change is required because our button semantics are different. Our cars DO NOT have the traditional overloaded res/accel or set/coast buttons; the Res and Set buttons behave exactly as labeled whether momentarily pressed or held.

Reply:
I realize this may be confusing so let me try to add some more clarity around VW driver expectations from their stock ACC:

* Pressing RES should not accelerate (this will be a very useful no-op for night-time DM reset instead of bumping the steering)
* Pressing SET should not decelerate (doesn't matter that much, but still)
* Pressing INCREASE or DECREASE to change the ACC setpoint should not engage ACC if it's disengaged (this is done all the time in stops or traffic jams)

We can't use accelCruise and decelCruise because the controlsd code treats them as (engage if disengaged) AND (resume or set) AND (accel or coast) functions all in one. We need to teach controlsd about these buttons and their separated functions.

Reply:
I'm happy to change the behavior in controls so you don't have to double define accelCruise and increaseCruise:
- for cars like Honda and GM, in car interface, I'll send both accelCruise and resumeCruise when the button is pressed, for example.
- controlsd will engage on setCruise and resumeCruise, instead of accelCruise and decelCruise.

Makes sense?

Reply:
In the short term, using stock ACC, that meets my needs and is completely acceptable. I have pushed a commit to withdraw the increase and decrease definitions. In the long term, if we have OP take long control (and I fully intend to), there are a few semantics we'll need to verify.

**Test case:** A driver is cruising on the highway at their ACC setpoint of 65mph. They come upon a stop-and-go traffic jam, during which ACC has slowed the car to 15mph. The driver wants to bring down the ACC setpoint somewhat, so the car doesn't go bonkers accelerating if there's briefly a gap ahead, that another driver might merge into suddenly.

- Setpoint expectation on one touch of decrease: 64mph
- Setpoint expectation on hold of decrease: 60mph (jumps in 5mph increments per second
- What a car with overloaded set/coast semantics would do : 14mph

The same sort of scenarios can apply to accelerating as well as decelerating. However, **that's not a problem I need to solve immediately**, and I'm not even totally sure that's how OP long works today, and if it does, your proposal of sending both events together for vehicles with overloaded button semantics would allow us to solve it cleanly.

The increase/decrease description was also a way to use them for following distance adjustment in the future, matching the stock capability. I'm given to understand that's not in Comma's strategy at this time. I have some feelings about that, but they won't be relevant in the near term future, and this PR is not the appropriate vehicle to express them anyway.

TL;DR: I'm cool with your suggestion.